sentry-core/src/main/java/io/sentry/core/Hub.java

```diff
@@ -294,22 +293,18 @@ public void withScope(@NotNull ScopeCallback callback) {
           "Instance is disabled and this 'withScope' call is a no-op.");
     } else {
       pushScope();
-      try {
-        StackItem item = stack.peek();
-        if (item != null) {
-          try {
-            callback.run(item.scope);
-          } catch (Exception e) {
-            logIfNotNull(
-                options.getLogger(), SentryLevel.ERROR, "Error in the 'withScope' callback.", e);
-          }
-        } else {
+      StackItem item = stack.peek();
+      if (item != null) {
+        try {
+          callback.run(item.scope);
+        } catch (Exception e) {
           logIfNotNull(
-              options.getLogger(), SentryLevel.FATAL, "Stack peek was null when withScope");
+              options.getLogger(), SentryLevel.ERROR, "Error in the 'withScope' callback.", e);
         }
-      } finally {
-        popScope();
+      } else {
+        logIfNotNull(options.getLogger(), SentryLevel.FATAL, "Stack peek was null when withScope");
       }
+      popScope();
```

Comment:
this try/catch only wraps `peek` which only returns null if empty.
`popScope` should be out of a `finally` block and if needed, to have it's own try/catch.

> Uncaught exceptions inside a finalizer won't even print a warning

Reply:
same here, I removed the quoting as I mixed up the meaning, but the change still applies.

Reply:
I'm OK with the change but just to clarify:

I tend to leave stuff that must happen within the `try` block in a `finally` block also for clarity. 
We must keep in mind in future changes that a early `return` added in either `try` or `catch` blocks would break the code.

Reply:
@bruno-garcia got it, but it's a `void` method, an early return will break compilation or tests.

Reply:
@bruno-garcia got it, but it's a `void` method, an early return will break compilation or tests.

Reply:
@bruno-garcia Honestly I do not like driving the business logic inside `finally` blocks as it somehow breaks readability and control flow. I only use it to clean up critical resources in a safe way.

Reply:
@bruno-garcia I understand your point but I feel like driving business logic inside a `finally` block breaks readability and control flow. I tend to use it only to cleanup critical resources in a safe way.